### PR TITLE
remove ReadFileWithVerify from StorageAPI

### LINKED ISSUE #4946: Remove ReadFileWithVerify in StorageAPI
The StorageAPI provides two functions for reading data from a disk - one with and one without bitrot verification. However the ReadFile function (without bitrot verification) basically calls ReadFileWithVerify with an explicitly as verified marked BitrotVerifier.

This leads to some unnecessary code which is not needed. Whenever somebody wants to read data from a disk he/she must ensure, that the he/she uses ReadFileWithVerify on the first read. This leads to code like:
```
if !verifier.IsVerified() {
  _, err = ReadFileWithVerify(volume, path, offset, buffer, verifier)
} esle {
   _, err = ReadFile(volume, path, offset, buffer)
}
```
But the posix layer already takes care that the disk is not verified twice. So this kind of logic is not needed.

## Possible Solution
Remove ReadFileWithVerifiy and combine logic with ReadFile

## Context
General refactoring

### PR DESCRIPTION
**Pull request by aead**
## Description

This change removes the ReadFileWithVerify function from the
StorageAPI. The ReadFile was basically a redirection to ReadFileWithVerify.
This change removes the redirection and moves the logic of
ReadFileWithVerify directly into ReadFile.
This removes a lot of unnecessary code in all StorageAPI implementations.

## Motivation and Context

Fixes #4946

## How Has This Been Tested?
manually

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactoring

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/posix.go [hunk 1]
```diff
@@ -535,23 +535,12 @@ func (s *posix) ReadAll(volume, path string) (buf []byte, err error) {
 // If an EOF happens after reading some but not all the bytes,
 // ReadFull returns ErrUnexpectedEOF.
```
Comment (fwessels) [inline cmd/posix.go]: ReadFull -> ReadFile

File: cmd/posix.go [hunk 2]
```diff
@@ -535,23 +535,12 @@ func (s *posix) ReadAll(volume, path string) (buf []byte, err error) {
...
 //
+// If the BitrotVerifier is not nil or not verified read file
```
Comment (fwessels) [inline cmd/posix.go]: read file --> ReadFile

File: cmd/retry-storage_test.go [hunk 3]
```diff
@@ -312,11 +312,11 @@ func TestRetryStorage(t *testing.T) {
 		var buf2 = make([]byte, 5)
 		verifier := NewBitrotVerifier(SHA256, sha256Hash([]byte("Hello, World")))
 		var n int64
-		if n, err = disk.ReadFileWithVerify("existent", "path", 7, buf2, verifier); err != nil {
+		if n, err = disk.ReadFile("existent", "path", 7, buf2, verifier); err != nil {
 			t.Fatal(err)
 		}
 		if err != nil {
-			t.Error("Error in ReadFileWithVerify", err)
+			t.Error("Error in ReadFil with bitrot verification", err)
```
Comment (fwessels) [inline cmd/retry-storage_test.go]: ReadFil -> ReadFile

File: cmd/storage-rpc-server-datatypes.go [hunk 4]
```diff
@@ -59,30 +59,15 @@ type ReadFileArgs struct {
...
-	// Data buffer read from the path at offset.
-	Buffer []byte
 
 	// Algorithm used in bit-rot hash computation.
 	Algo BitrotAlgorithm
 
 	// Stored hash value used to compare with computed value.
 	ExpectedHash []byte
+
+	// Indicates wether the disk should be verified
```
Comment (fwessels) [inline cmd/storage-rpc-server-datatypes.go]: wether -> whether 
should be verified --> has already been verified

File: cmd/posix.go (hunk 1 above)
Comment (aead) [reply cmd/posix.go]: done

File: cmd/posix.go (hunk 2 above)
Comment (aead) [reply cmd/posix.go]: done

File: cmd/retry-storage_test.go (hunk 3 above)
Comment (aead) [reply cmd/retry-storage_test.go]: done

File: cmd/storage-rpc-server-datatypes.go (hunk 4 above)
Comment (aead) [reply cmd/storage-rpc-server-datatypes.go]: done, thanks

File: cmd/posix_test.go [hunk 5]
```diff
@@ -1183,8 +1183,8 @@ var posixReadFileWithVerifyTests = []struct {
 	{file: "myobject", offset: 1000, length: 1001, algorithm: BLAKE2b512, expError: nil},             // 15
 }
 
-// TestPosixReadFileWithVerify - tests the posix level
-// ReadFileWithVerify API. Only tests hashing related
+// TestPosixReadFile with bitrot verification - tests the posix level
+// ReadFile API with a BitrotVerifier. Only tests hashing related
 // functionality. Other functionality is tested with
 // TestPosixReadFile.
 func TestPosixReadFileWithVerify(t *testing.T) {
```
Comment (fwessels) [inline cmd/posix_test.go:1190]: Maybe drop WithVerify part of name of test case?

**Review by fwessels — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**